armada/utils/source.py

```diff
@@ -0,0 +1,66 @@
+import pygit2
+import requests
+import tarfile
+import tempfile
+import shutil
+from os import path
+from urlparse import urlparse
+
+from ..exceptions import source_exceptions
+
+def git_clone(repo_url, branch='master'):
+    '''
+    clones repo to a /tmp/ dir
+    '''
+
+    if repo_url == '':
+        raise source_exceptions.GitLocationException(repo_url)
+
+    _tmp_dir = tempfile.mkdtemp(prefix='armada', dir='/tmp')
+
+    try:
+        pygit2.clone_repository(repo_url, _tmp_dir, checkout_branch=branch)
+    except Exception:
+        raise source_exceptions.GitLocationException(repo_url)
+
+    return _tmp_dir
+
+def get_tarball(tarball_url):
+    tarball_path = download_tarball(tarball_url)
+    return extract_tarball(tarball_path)
+
+def download_tarball(tarball_url):
+    '''
+    Downloads a tarball to /tmp and returns the path
+    '''
+    try:
+        tarball_filename = path.basename(urlparse(tarball_url).path)
+        response = requests.get(tarball_url)
+        with open('/tmp/{}'.format(tarball_filename), 'wb') as f:
```

Comment:
my initial reasoning was to preserve the filename extension but after trying out tempfile, it seems it doesn't actually matter. I will switch this to use tempfile